FreeRTOS/FreeRTOS-Plus-TCP
# Fix failing test cases for DNS unit tests

Description
-----------
This PR fixes the failing test cases for DNS unit tests which were failing after the IPv6 integration changes. 

NOTE: The unit test: _`test_CreateSocket_bind_fail`_ from DNS_Networking unit test module has been removed as its not relevant to the `DNS_CreateSocket` function after IPv6 integration changes. The flow of binding is validated in a different test case (as part of the test cases for `FreeRTOS_gethostbyname`) instead.

Test Steps
-----------
```
cmake -S test/unit-test -B test/unit-test/build/ 
make -C test/unit-test/build/ all 
cd test/unit-test/build/
ctest -E system --output-on-failure --verbose
```

Related Issue
-----------
DNS unit test cases were failing due to the changes in the source code as part of the IPv6 integration

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
